src/transformers/pipelines/token_classification.py

```diff
@@ -140,44 +142,45 @@ def __call__(self, inputs: Union[str, List[str]], **kwargs):
               Only exists if the offsets are available within the tokenizer
         """
 
-        _inputs, offset_mappings = self._args_parser(inputs, **kwargs)
+        _inputs, offset_mappings, model_batch_size = self._args_parser(inputs, **kwargs)
 
         answers = []
 
-        for i, sentence in enumerate(_inputs):
+        with self.device_placement():
 
-            # Manage correct placement of the tensors
-            with self.device_placement():
-
-                tokens = self.tokenizer(
-                    sentence,
-                    return_attention_mask=False,
-                    return_tensors=self.framework,
-                    truncation=True,
-                    return_special_tokens_mask=True,
-                    return_offsets_mapping=self.tokenizer.is_fast,
-                )
-                if self.tokenizer.is_fast:
-                    offset_mapping = tokens.pop("offset_mapping").cpu().numpy()[0]
-                elif offset_mappings:
-                    offset_mapping = offset_mappings[i]
-                else:
-                    offset_mapping = None
+            batch_tokenized = self.tokenizer.batch_encode_plus(
+                _inputs,
+                return_attention_mask=False,
+                return_tensors=self.framework,
+                truncation=True,
+                padding=True,
+                return_special_tokens_mask=True,
+                return_offsets_mapping=self.tokenizer.is_fast,
+            )
+            if self.tokenizer.is_fast:
+                offset_mappings = batch_tokenized.pop("offset_mapping").cpu().numpy()
 
-                special_tokens_mask = tokens.pop("special_tokens_mask").cpu().numpy()[0]
+            batch_special_tokens_mask = batch_tokenized.pop("special_tokens_mask").cpu().numpy()
 
-                # Forward
-                if self.framework == "tf":
-                    entities = self.model(tokens.data)[0][0].numpy()
-                    input_ids = tokens["input_ids"].numpy()[0]
-                else:
-                    with torch.no_grad():
-                        tokens = self.ensure_tensor_on_device(**tokens)
-                        entities = self.model(**tokens)[0][0].cpu().numpy()
-                        input_ids = tokens["input_ids"].cpu().numpy()[0]
+            if self.framework == "tf":
+                if model_batch_size > 1:
+                    warnings.warn("The `model_batch_size` argument is not supported for Tensorflow models. Ignoring")
```

Comment:
You're actually breaking tensorflow, right ? because you're not checking all sentences anymore.